BUILD

```diff
@@ -2272,4 +2272,29 @@ grpc_cc_library(
     ],
 )
 
+grpc_cc_library(
+    name = "google_protobuf",
+    srcs = [
+        "xds-generated/google/protobuf/any.upb.c",
```

Comment:
Instead of creating a new "xds-generated" top-level directory, can we simply put these files under third_party/upb/google/protobuf, so that they're in the same subdirectory as the files they're generated from?  In particular, keep in mind that these files may be used for something other than xds in the future.

Reply:
FWIW, note that the output files are not actually stored in the same directory as the source files, but the relative paths are the same.  For example, for a source file in src/core/lib/iomgr/foo.cc, the output file would be objs/src/core/lib/iomgr/foo.a (or maybe bazel-out instead of objs).

Reply:
I am not sure its third_party or third_party/upb where we cant have generated files.
the second path is obvious, its a submodule and I dont believe we want to commit there.

something like third_party/generated-files would probably work. 
We should syncup with @nicolasnoble to identify the correct location.

Reply:
Oh, right -- I forgot that we want to commit the output files to the repo in this case.

How about something like src/core/ext/google_protobuf_generated for the google protobuf files, and then put the xds-specific ones in src/core/ext/filters/client_channel/lb_policy/xds?

Reply:
Okay, if they all need to be in the same tree, then how about src/core/ext/upb-generated?